website/client/src/components/admin-panel/index.vue

```diff
@@ -0,0 +1,754 @@
+<template>
+  <div
+    v-if="user.contributor.admin"
```

Comment:
is this v-if necessary? if we add a redirect when not an admin I think it can be removed

Reply:
I don't think that would be safe enough. If a future change puts a bug into the router redirection code, anyone would be able to see and partially use the admin panel. The API would prevent them from viewing private data for any other user and would prevent them from making changes they're not authorised for BUT that would not not be obvious to them. It would appear to them that they had obtained access to a feature that totally breaks user privacy. The negative publicity could be very damaging if it spread widely.

Even though the chance of such a bug happening may be small, it's not impossible and the consequences could be bad. I don't think it's worth the risk for the sake of removing a v-if. :)

Reply:
I'm not sure I agree :) Yeah we could have a bug that removes the redirection check but I don't think it's a good reason to keep the extra check in place, as you said the API access would still be restricted and it would require a bunch of things to go wrong:

1. A bug in the redirection code
2. No one noticing it
3. Someone without admin privileges that manually types the url 

It's not like the extra check is a big deal to maintain but I don't think it's necessary or gives any advantage, we don't have the same check for all the app pages where login is necessary and as long as API data is protected (it is and even if it wasn't there would be other ways to access it) I think the chances of this being useful are so slim it's not worth it

Reply:
I don't think it's useful and usually I don't like having extra checks when not necessary but if you want to keep it that's fine